src/instrumentation/express/error_handler.ts

```diff
@@ -0,0 +1,31 @@
+import { Request, Response, NextFunction, ErrorRequestHandler } from "express"
+import * as opentelemetry from "@opentelemetry/api"
+
+export function expressErrorHandler(): ErrorRequestHandler {
+  return function (
+    err: Error & { status?: number },
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) {
+    const activeSpan = opentelemetry.trace.getSpan(
+      opentelemetry.context.active()
+    )
+    // if there's no `status` property, forward the error
+    // we also ignore client errors here
+    if (
+      activeSpan &&
+      err &&
+      (!err.status || (err.status && err.status >= 500))
+    ) {
+      if (activeSpan) {
+        activeSpan.recordException(err)
+        activeSpan.setStatus({
+          code: opentelemetry.SpanStatusCode.ERROR,
+          message: err.message
+        })
```

Comment:
Discussed this with @luismiramirez and we don't think it's a problem in practice, but just mentioning it for reference: in practice, the active span will be the HTTP span, and this status that is being set here will be overwritten with a `SpanStatusCode.OK` when the request is closed by the HTTP integration.

In practice, this doesn't matter to us, because we don't do anything with the span status. We just care about the exception event that is recorded. This is how you're supposed to record an exception, by recording the exception and setting the status -- if the status is later overriden by the instrumentation, that's out of our control.

Reply:
I thought you were going to create an additional child span for it. That would be useful if we later have an event timeline where you can see which error occurred where, if you have multiple errors.

Reply:
Recent processor changes don't support that behavior yet. This middleware doesn't know where an error happened because the active span is always the root span, so we don't know what middleware it comes from. Also, this middleware will catch the first HTTP error, and then the request will break, so it's unlikely that more than one error raises within the same trace.

Reply:
Luismi and I have discussed a very un-cooked, future nice-to-have idea on how to make the error appear on the span that raised it.

The gist of it is, instead of having this error middleware, having functionality inside the Express instrumentation that intercepts calls to `next(err)` in middlewares and request handlers. From a cursory look at the Express internals, it seems that `next` is always called with the error, both when an error is explicitly passed to `next` in the request handler, and implicitly in the Express internals when an error is thrown in the handler.

But for now, I think the error middleware will do. Problem with it is, of course, that it can't know where the error came from in the middleware chain.